src/Stack/Build/Source.hs

```diff
@@ -183,16 +182,18 @@ parseTargetsFromBuildOpts needTargets bopts = do
         (bcExtraDeps bconfig)
         (catMaybes $ Map.keys $ boptsFlags bopts)
 
-    (cliExtraDeps, targets) <-
+    let extraDeps' = flagExtraDeps <> bcExtraDeps bconfig
+
+    (_cliExtraDeps, targets) <-
         parseTargets
             needTargets
             (bcImplicitGlobal bconfig)
             snapshot
-            (flagExtraDeps <> bcExtraDeps bconfig)
+            extraDeps'
             (fst <$> rawLocals)
             workingDir
             (boptsTargets bopts)
-    return (mbp0, cliExtraDeps <> flagExtraDeps, targets)
+    return (mbp0, extraDeps', targets)
```

Comment:
I'm not certain why this change would be necessary for this feature, and it seems like it's likely to introduce a regression.